src/firmware/shared/circular_buffer_test.cpp

```diff
@@ -0,0 +1,175 @@
+extern "C"
+{
+#include "firmware/shared/circular_buffer.h"
+
+#include "shared/constants.h"
+}
+
+#include <gtest/gtest.h>
+
+class CircularBufferTest : public testing::Test
+{
+   protected:
+    virtual void SetUp(void)
+    {
+        circular_buffer = circular_buffer_create(10);
```

Comment:
what happens if we call `circular_buffer_create(0)`?

Reply:
If 0 is taken, `circular_buffer_create` will now set size to be 1. A test is also added for this.

Reply:
why? just assert(size>0) or throw an exception